grtc/app.js

```diff
@@ -0,0 +1,471 @@
+'use strict';
+
+import Peer from 'simple-peer';
+import uuidv4 from 'uuid/v4';
+import uuidv1 from 'uuid/v1';
+import { EventEmitter } from 'events';
+
+import crypto from './crypto';
+
+/**
+ * Signal class that uses promise not event emitters and can be used standalone.
+ * getSignal and updateSignal lets you update and fetch data regarding the shared key.
+ */
+class Signal {
+	/**
+	 * @param {string} url url to which signal should go.
+	 * @param {string} grtcID global id shared by url
+	 * @param {string} peerID global peerID that maps to username.
+	 * @param {string} peerName global username that maps to peerID.
+	 * @param {object} signalID signal generated by peer
+	 * url is base url of page.
+	 * grtcID is collaborate param from url.
+	 * paramID is changing peerID on refresh or every new grtc instance.
+	 * signalID is peer signal used to traverse and connect P2P.
+	 */
+	constructor( url, grtcID, peerID, peerName, signalID ) {
+		const self = this;
+		self.url = url;
+		self.grtcID = grtcID;
+		self.peerID = peerID;
+		self.signalID = signalID;
+		self.peerName = peerName;
+	}
+
+	/**
+	 * Clear the key forcefully in kv.
+	 * @return {promise} promise object
+	 */
+	clearSignal() {
+		const self = this;
```

Comment:
Arrow functions preserve `this` reference, you don't need to create a separate variable for tracking `this`.

Reply:
Yes, removed all `self` occurrences.